src/Microsoft.Azure.WebJobs.Logging/Internal/TimeBucket.cs

```diff
@@ -27,5 +29,63 @@ public static long ConvertToBucket(DateTime dt)
             }
             return min;
         }
+
+        // Used to compute epoch. 
+        const long ticksBaseline = 633663648000000000L;
+        const long ticksPerMonth = 26785144663987L;
+        
+        public static DateTime CommonEpoch = new DateTime(0, DateTimeKind.Utc); // maps to epoch 0 
+        public static string CommonEpochSuffix = "common";
+
+        public static long GetEpochNumberFromTable(CloudTable table)
+        {
+            string tableName = table.Name;
+            if (tableName.Length < 5)
+            {
+                return -1;
+            }
+            string suffix = tableName.Substring(tableName.Length - 5, 5);
+            long epoch;
+            if (long.TryParse(suffix, out epoch))
+            {
+                return epoch;
+            }
+            return -1;
+        }
+
+        // Epoch must be positive, orderd integers.
+        // Use YYYYMM 
+        private static long GetEpochNumber(DateTime epoch)
+        {
+            var year = epoch.Year; // 4 digit
+            var month = epoch.Month; // 1..12
+
+            int i = year * 100 + month;
+            return i;
+        }
+
+        public static CloudTable GetTableForEpoch(this ILogTableProvider tableLookup, DateTime epoch)
```

Comment:
My intention was to make the interface as simple as possible (and hence do as much possible before making the call). The only thing the interface absolutely *has* to do is the suffix--> table mapping.   
But as you're alluding to, that means the interface can't control the policy (group by month, etc).  I can shift this around. 

---
In reply to: [82906498](https://github.com/Azure/azure-webjobs-sdk/pull/862#discussion_r82906498) [](ancestors = 82906498)